packages/dds/shared-object-base/src/serializer.ts

```diff
@@ -225,7 +228,16 @@ export class FluidSerializer implements IFluidSerializer {
 		handle: IFluidHandleInternal,
 		bind: ISharedObjectHandle,
 	): ISerializedHandle {
-		bind.bind(handle);
+		if (this.forSummarizer) {
+			assert(bind.isAttached, "Expected bind source to be attached in the Summarizer");
```

Comment:
I think this should be safe. But should we consider adding this behind a feature flag or something to be sure?